contrib/src/test/java/com/datatorrent/contrib/avro/AvroFileToPojoModuleTest.java

```diff
@@ -0,0 +1,260 @@
+      String className = description.getClassName();
+      this.dir = "target/" + className + "/" + methodName;
+      Attribute.AttributeMap attributes = new Attribute.AttributeMap.DefaultAttributeMap();
+      attributes.put(Context.DAGContext.APPLICATION_PATH, dir);
+      context = mockOperatorContext(1, attributes);
+      Attribute.AttributeMap portAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+      portAttributes.put(Context.PortContext.TUPLE_CLASS, SimpleOrder.class);
+      portContext = new TestPortContext(portAttributes);
+    }
+
+    @Override
+    protected void finished(Description description)
+    {
+      try {
+        FileUtils.deleteDirectory(new File("target/" + description.getClassName()));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+
+  private void createAvroInput(int cnt)
+  {
+    recordList = Lists.newArrayList();
+
+    while (cnt > 0) {
+      GenericRecord rec = new GenericData.Record(new Schema.Parser().parse(AVRO_SCHEMA));
+      rec.put("orderId", cnt * 1L);
+      rec.put("customerId", cnt * 2);
+      rec.put("total", cnt * 1.5);
+      rec.put("customerName", "*" + cnt + "*");
+      cnt--;
+      recordList.add(rec);
+    }
+  }
+
+  private void writeAvroFile(File outputFile)
+  {
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(new Schema.Parser().parse(AVRO_SCHEMA));
+    try (DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter)) {
+      dataFileWriter.create(new Schema.Parser().parse(AVRO_SCHEMA), outputFile);
+
+      for (GenericRecord record : recordList) {
+        dataFileWriter.append(record);
+      }
+      FileUtils.moveFileToDirectory(new File(outputFile.getAbsolutePath()), new File(testMeta.dir), true);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testAvroToPojoModule() throws Exception
+  {
+    try {
+      FileContext.getLocalFSFileContext().delete(new Path(new File(testMeta.dir).getAbsolutePath()), true);
+      int cnt = 7;
+      createAvroInput(cnt);
+      writeAvroFile(new File(FILENAME));
+      createAvroInput(cnt - 2);
+      writeAvroFile(new File(OTHER_FILE));
+
+      avroFileToPojoModule.setAvroFileDirectory(testMeta.dir);
+      avroFileToPojoModule.setPojoClass(SimpleOrder.class);
+
+      AvroToPojo avroToPojo = new AvroToPojo();
+      avroToPojo.setPojoClass(SimpleOrder.class);
+
+      EmbeddedAppLauncherImpl lma = new EmbeddedAppLauncherImpl();
+      Configuration conf = new Configuration(false);
+
+      AvroToPojoApplication avroToPojoApplication = new AvroToPojoApplication();
+      avroToPojoApplication.setAvroFileToPojoModule(avroFileToPojoModule);
+
+      lma.prepareDAG(avroToPojoApplication, conf);
+      EmbeddedAppLauncherImpl.Controller lc = lma.getController();
+      lc.run(10000);// runs for 10 seconds and quits
```

Comment:
Since you want to verify that the operator is working, you need to check the results and the test only needs to run until these results are available. For use of non-deprecated  API and wait until results example, see https://github.com/apache/apex-malhar/blob/master/examples/exactly-once/src/test/java/org/apache/apex/examples/exactlyonce/ExactlyOnceJdbcOutputTest.java#L118